### PR TITLE
Support passing blocksize to PoolManager

### LINKED ISSUE #2067: PoolManager does not support argument blocksize
### Subject

urllib3's PoolManager does not support `blocksize`. This seems to be a new argument accepted by Python's standard library's http clients since 3.7 (https://docs.python.org/3.6/library/http.client.html vs https://docs.python.org/3.7/library/http.client.html)

Adding `key_blocksize` to poolmanager.py's `_key_fields` fixes the issue but it probably should only be appended when it is actually supported.

### Environment

OS Linux-5.4.0-47-generic-x86_64-with-glibc2.29
Python 3.8.5
urllib3 1.26.2

### Steps to Reproduce

```
from urllib3 import PoolManager
with PoolManager(blocksize = 8192) as pool:
    response = pool.urlopen("GET", "https://www.google.com")
```

### Expected Behavior

Should perform a GET request successfully.

### Actual Behavior

The following exception is thrown

```
Traceback (most recent call last):
  File "report.py", line 13, in <module>
    response = pool.urlopen("GET", "https://www.google.com")
  File "/home/geancarlo/redacted/poc/venv/lib/python3.8/site-packages/urllib3/poolmanager.py", line 364, in urlopen
    conn = self.connection_from_host(u.host, port=u.port, scheme=u.scheme)
  File "/home/geancarlo/redacted/poc/venv/lib/python3.8/site-packages/urllib3/poolmanager.py", line 245, in connection_from_host
    return self.connection_from_context(request_context)
  File "/home/geancarlo/redacted/poc/venv/lib/python3.8/site-packages/urllib3/poolmanager.py", line 258, in connection_from_context
    pool_key = pool_key_constructor(request_context)
  File "/home/geancarlo/redacted/poc/venv/lib/python3.8/site-packages/urllib3/poolmanager.py", line 124, in _default_key_normalizer
    return key_class(**context)
TypeError: __new__() got an unexpected keyword argument 'key_blocksize'
```

### PR DESCRIPTION
**Pull request by imkaka**
Closes #2067 

Heavily inspired by #2081 and the work done by @geancarlo and the PR comments.
Please let me know if any improvements are required.

### CODE REVIEW

File: test/test_poolmanager.py [hunk 1]
```diff
@@ -392,3 +399,30 @@ def test_pool_manager_no_url_absolute_form(self):
...
+        """Assert normalizer defaults to blocksize DEFAULT_BLOCKSIZE(16384)"""
+        p = PoolManager()
+
+        pool_blocksize_default = p.connection_from_url(
+            "http://example.com", {"blocksize": DEFAULT_BLOCKSIZE}
+        )
+        assert pool_blocksize_default.conn_kw["blocksize"] == DEFAULT_BLOCKSIZE
+
+        pool_blocksize_not_set = p.connection_from_url("http://example.com")
+        assert pool_blocksize_not_set == pool_blocksize_default
```
Comment (V1NAY8) [inline test/test_poolmanager.py:413]: I am thinking, can we use `pytest.parametrize` to parametrize the `blocksize` and have a single method that can run all the 3 tests ?
Probably do these asserting on all the three cases
```
        assert pool_blocksize_8192.conn_kw["blocksize"] == blocksize
        assert pool_blocksize_8192._get_conn().blocksize == blocksize
```
Reference: https://docs.pytest.org/en/latest/how-to/parametrize.html

Comment (imkaka) [reply test/test_poolmanager.py:413]: Makes sense, will do the refactor.

**Review by sethmlarson — DISMISSED** [review]
Thanks for this! Going to try only setting the `blocksize` within the key normalizer, in theory that'll work just fine.

**Comment by imkaka** [comment]
> Thanks for this! Going to try only setting the `blocksize` within the key normalizer, in theory that'll work just fine.



> Thanks for this! Going to try only setting the `blocksize` within the key normalizer, in theory that'll work just fine.

Doing this is resulting in the failure of `(None, _DEFAULT_BLOCKSIZE)` test case, where the user explicitly setting `blocksize` to `None`.

```
pool_blocksize = p.connection_from_url(
            "http://example.com", {"blocksize": input_blocksize}
        )
```
What should we do? remove this case or revert the change or anything else @sethmlarson?

**Comment by sethmlarson** [comment]
@imkaka Thanks for your patience on this, I saw the failure but haven't had a chance to debug it yet. Hopefully will get to this today :)

**Review by sethmlarson — APPROVED** [review]
I think that extra check there is for the best, we'll keep it for now. Thanks for this! Will merge on CI passing.

**Review by sethmlarson — APPROVED** [review]
LGTM

**Pull request merged**